src/modules/order/components/ShippingInfo.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import {
+  useForm,
+  SubmitHandler,
+  SubmitErrorHandler,
+  DeepMap,
+  FieldError,
+} from 'react-hook-form';
+
+import { useSelector } from 'react-redux';
+import { AppState } from 'modules/redux-store/models';
+
+import { useFirebaseHooks } from 'modules/firebase/hooks';
+
+import { OrderAction } from 'modules/order/redux';
+import { useDispatch } from 'react-redux';
+
+import { useHistory } from 'react-router';
+
+import { useOrder } from 'modules/order/hooks';
+export const ShippingInfo: React.FC = () => {
+  const { pushOrder } = useFirebaseHooks('orders');
+
+  const { toppings, size, discount, quantity } = useSelector(
+    (state: AppState) => state.orderReducer,
+  );
+  const { user } = useSelector((state: AppState) => state.authReducer);
+
+  type FormValues = {
+    address: string;
+    city: string;
+    postalCode: number;
+    country: string;
+  };
+
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const { getCurrentPrice } = useOrder();
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+  } = useForm<FormValues>();
+
+  const onSubmit: SubmitHandler<FormValues> = (data: FormValues) => {
+    console.log('data', data);
+    dispatch(OrderAction.address(data.address));
+    dispatch(OrderAction.postalCode(data.postalCode));
+    dispatch(OrderAction.city(data.city));
+    dispatch(OrderAction.country(data.country));
+    pushOrder({
+      user: user?.email,
+      address: data.address,
+      city: data.city,
+      postalCode: data.postalCode,
+      country: data.country,
+      price: getCurrentPrice(),
+      size: size,
+      quantity: quantity,
```

Comment:
ako ti se prop zove isto kao i druga varijabla, onda to možeš skratiti tako što ćeš napisati samo `quantity, size, discount`